src/guc.c

```diff
@@ -1105,6 +1121,20 @@ _guc_init(void)
 							 /* show_hook= */ NULL);
 #endif
 
+	DefineCustomEnumVariable(MAKE_EXTOPTION("default_hypercore_use_access_method"),
```

Comment:
Should we use a boolean value instead? It is in effect a boolean, so why not?

Reply:
I'm not entirely convinced why this GUC is not a simple boolean instead of this enum. Looks like internally we're already using bool type to deal with it. 

And what's the point to set it to "none"?? I think it only make the UX a bit confuse. A regular boolean here seems to be more natural.

Reply:
Ok. I will add a commit to see what changes needs to be made to use a boolean.

Reply:
Switched to use boolean as a separate commit. PTAL. There is an extra error message printed, so will see if that can be eliminated.

Reply:
The extra error message is sensible, so will not remove it. It is a result from the fact that we do not have a "null" value for the default, which means that if you try to recompress a partially compressed hypercore table with "use_access_method => false" it will do a recompression instead.